qa/common/gen_qa_reshape_models.py

```diff
@@ -493,6 +518,116 @@ def create_ensemble_modelconfig(
+        return res
+        
+    normalize_variable_shape.idx = 0
+
+    assert len(input_shapes) == len(output_shapes)
+    if not tu.validate_for_onnx_model(dtype, dtype, dtype,
+                                      input_shapes[0], input_shapes[0], input_shapes[0]):
+        return
+
+    onnx_dtype = np_to_onnx_dtype(dtype)
+    io_cnt = len(input_shapes)
+
+    # Create the model
+    model_name = tu.get_zero_model_name("onnx_nobatch" if max_batch == 0 else "onnx",
+                                   io_cnt, dtype)
+    model_version_dir = models_dir + "/" + model_name + "/" + str(model_version)
+
+    batch_dim = [] if max_batch == 0 else [max_batch]
+
+    onnx_nodes = []
+    onnx_inputs = []
+    onnx_outputs = []
+    for io_num in range(io_cnt):
+        # Repeat so that the variable dimension name is different
+        in_shape = normalize_variable_shape(input_shapes[io_num])
+        out_shape = normalize_variable_shape(output_shapes[io_num])
+        in_name = "INPUT{}".format(io_num)
+        out_name = "OUTPUT{}".format(io_num)
+        out_shape_name = out_name + "_shape"
+
+        onnx_inputs.append(onnx.helper.make_tensor_value_info(in_name, onnx_dtype, batch_dim + in_shape))
+        onnx_outputs.append(onnx.helper.make_tensor_value_info(out_name, onnx_dtype, batch_dim + out_shape))
+        
+        if input_shapes == output_shapes:
+            onnx_nodes.append(onnx.helper.make_node("Identity", [in_name], [out_name]))
+        else:
+            onnx_nodes.append(onnx.helper.make_node("Shape", [out_name], [out_shape_name]))
+            onnx_nodes.append(onnx.helper.make_node("Reshape", [in_name, out_shape_name], [out_name]))
+
+    graph_proto = onnx.helper.make_graph(onnx_nodes, model_name, onnx_inputs, onnx_outputs)
+    model_def = onnx.helper.make_model(graph_proto, producer_name="TRTIS")
+
+    try:
+        os.makedirs(model_version_dir)
+    except OSError as ex:
+        pass # ignore existing dir
+
+    onnx.save(model_def, model_version_dir + "/model.onnx")
+
+
+def create_onnx_modelconfig(
+        models_dir, model_version, max_batch, dtype,
+        input_shapes, input_model_shapes, output_shapes, output_model_shapes):
+
+    assert len(input_shapes) == len(input_model_shapes)
+    assert len(output_shapes) == len(output_model_shapes)
+    assert len(input_shapes) == len(output_shapes)
+    if not tu.validate_for_onnx_model(dtype, dtype, dtype,
+                                      input_shapes[0], input_shapes[0], input_shapes[0]):
+        return
+
+    io_cnt = len(input_shapes)
+
+    # Use a different model name for the non-batching variant
+    model_name = tu.get_zero_model_name("onnx_nobatch" if max_batch == 0 else "onnx",
+                                   io_cnt, dtype)
+    config_dir = models_dir + "/" + model_name
+    
+    # Must make sure all Onnx models will be loaded to the same GPU if they are
+    # run on GPU. This is due to the current limitation of Onnx Runtime
```

Comment:
Can you link to the issue tracking this on onnx site

Reply:
Create an issue referring to the one on onnx site or just put the link inline?

Reply:
Just inline the link to the existing onnx runtime issue.